### PR TITLE
Signing-related cleanup in pkg/config

### PR DESCRIPTION
**Pull request by wlahti**
#### Type of change

- Improvement (improvement to code, performance, etc)

#### Description

Code cleanup/improvements to signing-related functions. Also, fix a data race in signer_test.go.

#### Related issues

[FAB-17529](https://jira.hyperledger.org/browse/FAB-17529)

### CODE REVIEW

File: pkg/config/signer.go [hunk 1]
```diff
@@ -7,155 +7,53 @@ SPDX-License-Identifier: Apache-2.0
...
-// SigningIdentity is an MSP Identity that can be used to sign configuration updates and transaction.
+// SigningIdentity is an MSP Identity that can be used to sign configuration
+// updates.
 type SigningIdentity struct {
-	pemCert    []byte // pem encoded publicCert
-	mspID      string
-	privateKey *ecdsa.PrivateKey
-	publicKey  *ecdsa.PublicKey
+	Certificate *x509.Certificate
+	PrivateKey  crypto.PrivateKey
```
Comment (DereckLuo) [inline pkg/config/signer.go:23]: crypto.PrivateKey is an empty interface. Should we define privateKey type as crypto.Signer? as ecdsa.privateKey implement crypto.Signer.

**Review by DereckLuo — COMMENTED** [review]
func (s *SigningIdentity) Public was removed from previous PR. We might want to add this in, so SigningIdentity will implement crypto.Signer.

**Comment by sykesm** [comment]
Looks like a success notification was dropped from azure pipelines.

File: pkg/config/signer.go (hunk 1 above)
Comment (sykesm) [reply pkg/config/signer.go:23]: the interface is good here; while we really only handle ecdsa, it more accurately describes the dependency

**Review by sykesm — DISMISSED** [review]

**Review by sykesm — APPROVED** [review]

**Pull request merged**